### PR TITLE
Fix FP S1128 (`unused-imports`): Vue.js imports used in templates

### LINKED ISSUE #3484: Fix FP S1128 (`unused-imports`): Vue.js imports used in templates
The script attribute `lang="ts"` is necessary for this FP to occur.

```javascript
<template>
  <Foo />
</template>

<script setup lang="ts">
import Foo from './Foo' // FP
</script>
```

Reported in:
- https://community.sonarsource.com/t/remove-this-unused-import-of-xxx-rule-is-not-detecting-them-in-template-vue-js/74330
- https://community.sonarsource.com/t/false-postive-javascript-s1128-with-vue-script-setup-lang-ts/73990
- https://community.sonarsource.com/t/vue3-composition-api-and-script-setup-syntax-reports-unused-imports-since-october-22/74614

### PR DESCRIPTION
**Pull request by ilia-kebets-sonarsource**
Fixes #3484

### CODE REVIEW

**Comment by ilia-kebets-sonarsource** [comment]
@yassin-kammoun-sonarsource I verified that putting back the code removed in that commit does indeed fix the FP: https://github.com/SonarSource/SonarJS/commit/03dc60ab105e0133e9d3b096038a38239a896e96

File: eslint-bridge/tests/tools/testers/comment-based/helpers/comments.ts [hunk 1]
```diff
@@ -29,11 +29,14 @@ export interface Comment {
...
-  const parsed = buildSourceCode(
-    { fileContent, filePath: '', fileType: null, tsConfigs: [] },
-    null,
-  );
+/**
+ *
+ * @param fileContent
+ * @param filePath used to know if it's Vue or not
+ * @returns
+ */
```
Comment (yassin-kammoun-sonarsource) [inline eslint-bridge/tests/tools/testers/comment-based/helpers/comments.ts:37]:
```suggestion
```

Comment (yassin-kammoun-sonarsource) [reply eslint-bridge/tests/tools/testers/comment-based/helpers/comments.ts:37]: I don't think we need to duplicate this comment, especially because we can track back the change to the commit which introduced the change.

File: eslint-bridge/tests/tools/testers/comment-based/helpers/file.ts [hunk 2]
```diff
@@ -37,9 +37,10 @@ export class FileIssues {
   /**
    * Parses the file into its expected errors. Throws if error flags are not well formatted.
    * @param fileContent
+   * @param filePath used to know if it's Vue or not
```
Comment (yassin-kammoun-sonarsource) [inline eslint-bridge/tests/tools/testers/comment-based/helpers/file.ts:40]:
```suggestion
   * @param filePath
```

**Review by yassin-kammoun-sonarsource — CHANGES_REQUESTED** [review]
Lookgs good to me. However, could you update the definition of `languageFromFilePath` and take into account `.vue` files:

https://github.com/SonarSource/SonarJS/blob/ff94357392bc2b742ce4cdfe0a0d6d7ce2d836be/eslint-bridge/tests/tools/testers/comment-based/launcher.test.ts#L110-L117

You will need to pass the `fileContent` from `parseForESLint` and use this [regular expression](https://github.com/SonarSource/SonarJS/blob/d6b6ca841ef300235775bd611107bdf64e4117ab/sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptFilePredicate.java#L32) to make the decision on the language of such a file.

There won't be a major impact, but by doing so, the comment-based testing framework will be closer to a `.vue` file analysis in production.

File: eslint-bridge/tests/tools/testers/comment-based/helpers/comments.ts (hunk 1 above)
Comment (ilia-kebets-sonarsource) [reply eslint-bridge/tests/tools/testers/comment-based/helpers/comments.ts:37]: ack

File: eslint-bridge/tests/tools/testers/comment-based/launcher.test.ts [hunk 3]
```diff
@@ -105,11 +109,15 @@ export function parseForESLint(
 }
 
 /**
- * Returns the source code's language based on the file path.
+ * Returns the source code's language based on the file content and path.
  */
-function languageFromFilePath(filePath: string): Language {
+function languageFromFile(fileContent: string, filePath: string): Language {
+  const hasScriptTagWithLangTs = /\\<script[.\\s]*lang=['\"]ts['\"][.\\s]*\\>/;
```
Comment (yassin-kammoun-sonarsource) [inline eslint-bridge/tests/tools/testers/comment-based/launcher.test.ts:115]: The regex doesn't seem to work. Because we are using a JavaScript regex literal here, the semantics is different from Java, and we can't copy-paste the regex because we don't need to escape characters.

Also, I realize now that the regex in the file predicate (as well as this one) does not support `<script setup...`. Let's fix the regex of `JavaScriptFilePredicate.java` in a different PR or have the change in this one but in a dedicated commit (which we should not squash when the time to merge comes).

Long story short, the regex should include `setup` and mark it as optional.

**Review by yassin-kammoun-sonarsource — CHANGES_REQUESTED** [review]

**Comment by ilia-kebets-sonarsource** [comment]
@yassin-kammoun-sonarsource LMK if you're fine with the refactoring of the regex. If yes, I'll rebase this branch into 3 commits:
- Support VueJS test files in comment-based framework
- Fix FP S1128 (`unused-imports`): Vue.js imports used in templates on TS script
- Fix VueJS typescript filter

You can check the pre-last commit for another variant of the regex, but I prefer the one I settled on:
- the last: `/<script[^>]+lang=['"]ts['"][^>]*>/`
- the pre-last: `/<script[\s\S]*?lang=['"]ts['"][\s\S]*?>/`

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptFilePredicate.java [hunk 4]
```diff
@@ -29,7 +29,7 @@
 
 public class JavaScriptFilePredicate {
 
-  private static final String regex = "\\<script[.\\s]*lang=['\"]ts['\"][.\\s]*\\>";
+  private static final String regex = "\\<script[^>]+lang=['\"]ts['\"][^>]*>";
```
[the remark this answers is not in the document]
Comment (ilia-kebets-sonarsource) [reply sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptFilePredicate.java:32]: @yassin-kammoun-sonarsource I have added 3 more test cases to illustrate my changes.
